service/javadsl/server/src/main/scala/com/lightbend/lagom/internal/javadsl/server/AdditionalRoutersProvider.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.javadsl.server
+
+import java.util
+
+import akka.annotation.InternalApi
+import com.lightbend.lagom.javadsl.server.AdditionalRouter
+import javax.inject.{ Inject, Provider }
+import play.api.inject.Injector
+import play.api.routing.Router
+
+/**
+ * Provides a list of [[play.routing.Router]]s built from a list of [[AdditionalRouter]]
+ *
+ * @param injector
+ * @param additionalRouters
+ */
+@InternalApi
+private[lagom] class AdditionalRoutersProvider @Inject() (
+  injector:          Injector,
+  additionalRouters: util.List[AdditionalRouter]
```

Comment:
So here it's a list of `AdditionalRouter`, but when binding I saw `new TypeLiteral<List<Router>>` - apparently Guice is smart enough to work that out? Might be worth considering wrapping the list in a class just to make it easier to follow? No big deal since it's internal API though

Reply:
`TypeLiteral` is Guice's way of working around erasure of type parameters https://github.com/google/guice/wiki/FrequentlyAskedQuestions#how-to-inject-class-with-generic-type

Reply:
Right - with a wrapper (e.g. an 'AdditionalRouters` class containing a list of `AdditionalRouter`s) you wouldn't need that feature, and have a somewhat easier time discovering the bindings in an IDE. But it's fine like this ;)